core/src/apps/management/recovery_device/__init__.py

```diff
@@ -31,68 +33,75 @@ async def recovery_device(msg: RecoveryDevice) -> Success:
-    if dry_run and not storage_device.is_initialized():
-        raise wire.NotInitialized("Device is not initialized")
+    if recovery_kind == RecoveryKind.NormalRecovery:
+        if storage_device.is_initialized():
+            raise wire.UnexpectedMessage("Already initialized")
+    elif recovery_kind in (RecoveryKind.DryRun, RecoveryKind.UnlockRepeatedBackup):
+        if not storage_device.is_initialized():
+            raise wire.NotInitialized("Device is not initialized")
+        # check that only allowed fields are set
+        for key, value in msg.__dict__.items():
+            if key not in DRY_RUN_ALLOWED_FIELDS and value is not None:
+                raise wire.ProcessError(
+                    f"Forbidden field set in dry-run: {key}"
+                )
+    else:
+        raise ValueError("Unknown RecoveryKind")
+
     if msg.enforce_wordlist is False:
         raise wire.ProcessError(
             "Value enforce_wordlist must be True, Trezor Core enforces words automatically."
         )
-    if dry_run:
-        # check that only allowed fields are set
-        for key, value in msg.__dict__.items():
-            if key not in DRY_RUN_ALLOWED_FIELDS and value is not None:
-                raise wire.ProcessError(f"Forbidden field set in dry-run: {key}")
     # END validate
     # --------------------------------------------------------
 
     if storage_recovery.is_in_progress():
         return await recovery_process()
 
-    # --------------------------------------------------------
-    # _continue_dialog
-    if not dry_run:
+    if recovery_kind == RecoveryKind.NormalRecovery:
         await confirm_reset_device(TR.recovery__title_recover, recovery=True)
-    else:
-        await confirm_action(
-            "confirm_seedcheck",
-            TR.recovery__title_dry_run,
-            description=TR.recovery__check_dry_run,
-            br_code=ButtonRequestType.ProtectCall,
-            verb=TR.buttons__check,
-        )
-    # END _continue_dialog
-    # --------------------------------------------------------
 
-    if not dry_run:
         # wipe storage to make sure the device is in a clear state
         storage.reset()
 
-    # for dry run pin needs to be entered
-    if dry_run:
-        curpin, salt = await request_pin_and_sd_salt(TR.pin__enter)
-        if not config.check_pin(curpin, salt):
-            await error_pin_invalid()
-
-    if not dry_run:
         # set up pin if requested
         if msg.pin_protection:
             newpin = await request_pin_confirm(allow_cancel=False)
             config.change_pin("", newpin, None, None)
 
         storage_device.set_passphrase_enabled(bool(msg.passphrase_protection))
+
         if msg.u2f_counter is not None:
             storage_device.set_u2f_counter(msg.u2f_counter)
+
         if msg.label is not None:
             storage_device.set_label(msg.label)
+    elif recovery_kind in (RecoveryKind.DryRun, RecoveryKind.UnlockRepeatedBackup):
```

Comment:
add a separator line
```suggestion

    elif recovery_kind in (RecoveryKind.DryRun, RecoveryKind.UnlockRepeatedBackup):
```

Reply:
https://github.com/trezor/trezor-firmware/pull/3743/commits/1898605cc75681e72f36e38b9d352d3589988e59#diff-1751448e00180998cfe48dbfeb7a35ffb20d53c8938f071a5b44eb456a5f8943R86